File: output/csbc_semish_2026/gemini_3_1_pro_preview/few/cpp/Arquivos.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    long long b;
    if (cin >> n >> b) {
        vector<long long> files(n);
        for (int i = 0; i < n; i++) {
            cin >> files[i];
        }

        sort(files.begin(), files.end());

        int left = 0;
        int right = n - 1;
        int folders = 0;

        while (left <= right) {
            if (left == right) {
                folders++;
                break;
            }
            if (files[left] + files[right] <= b) {
                folders++;
                left++;
                right--;
            } else {
                folders++;
                right--;
            }
        }

        cout << folders << "\n";
    }

    return 0;
}

File: output/csbc_semish_2026/gemini_3_1_pro_preview/few/python/Arquivos.py
import sys

input_data = sys.stdin.read().split()
if not input_data:
    exit()

n = int(input_data[0])
b = int(input_data[1])
files = [int(x) for x in input_data[2:2+n]]

files.sort()

left = 0
right = n - 1
folders = 0

while left <= right:
    if left == right:
        folders += 1
        break
    if files[left] + files[right] <= b:
        left += 1
        right -= 1
        folders += 1
    else:
        right -= 1
        folders += 1

print(folders)